geonode/static/geonode/js/upload/LayerInfo.js

```diff
@@ -706,10 +706,12 @@ define(function (require, exports) {
                     layer_name = target.data('layer'),
                     file_name  = target.data('file');
                 self.removeFile(file_name);
+                if (self.files.length == 0) {
```

Comment:
Comment on L709: Detects if the above call to `removeFile` has removed the final file associated with the LayerInfo (length == 0). If so, delete this LayerInfo object from the list of stored LayerInfo objects that have been staged for upload. (Previously this was not being done, trapping the user in an error state unless they clicked "Clear")